services/provider/server/server.go

```diff
@@ -681,12 +707,31 @@ func (s *OCSProviderServer) GetStorageClaimConfig(ctx context.Context, req *pb.S
 				"clusterID":          getSubVolumeGroupClusterID(subVolumeGroup),
 				"subvolumegroupname": subVolumeGroup.Name,
 				"fsName":             subVolumeGroup.Spec.FilesystemName,
-				"pool":               subVolumeGroup.GetLabels()[v1alpha1.CephFileSystemDataPoolLabel],
+				"pool":               subVolumeGroup.GetLabels()[ocsv1alpha1.CephFileSystemDataPoolLabel],
 				"csi.storage.k8s.io/provisioner-secret-name":       provisionerSecretName,
 				"csi.storage.k8s.io/node-stage-secret-name":        nodeSecretName,
 				"csi.storage.k8s.io/controller-expand-secret-name": provisionerSecretName,
 			}
 
+			clientProfileSpec := &csiopv1a1.ClientProfileSpec{
+				CephFs: &csiopv1a1.CephFsConfigSpec{
+					SubVolumeGroup: cephRes.Name,
+				},
+			}
```

Comment:
Let's build that after we have calculated the kernel mount options (or even hardcoded inside the ExternalResource definition) so there will be no need to edit the object we created.

Reply:
we will not be editing if there is no mountoption, I can move the same to after calculation.

Reply:
A map can be nil, so I don't understand the issue. 
Start be defining it as nil, then have your checks, then overwrite the nil if nececary, then build the struct with whatever is in the map's variable